### DISCUSSION TITLE
How to use with TypeScript

### DISCUSSION THREAD

**Discussion by lukepuplett** [js-fns/date-fns#2502] [Q&A] [answered]
See question on SO here:

https://stackoverflow.com/questions/67840725/how-do-i-use-date-fns-with-typescript

Your documentation says "Start using"

    import { formatDistance, subDays } from 'date-fns'

But this doesn't work from a TypeScript `.ts` source file, so I assume the syntax is wrong for TypeScript.

> error TS2307: Cannot find module 'lib/date-fns' or its corresponding type declarations.

While typing this, I've looked around and found this:

https://github.com/date-fns/date-fns/blob/master/examples/typescript/example.ts

So maybe the syntax is correct and there's some other problem.

**Comment by tan75** [comment, marked as answer]
Hi @lukepuplett 
If you have any other questions - let us know.

**Reply by lukepuplett** [reply]
I also posted to SO. I was having problems in general with TypeScript, JavaScript and NPM packages and modules in the browser.

https://stackoverflow.com/questions/67840725/how-do-i-use-date-fns-with-typescript/67903272#67903272
